transformer_engine/pytorch/module/linear.py

```diff
@@ -94,25 +102,30 @@ def forward(
         # pylint: disable=missing-function-docstring
 
         # Make sure input dimensions are compatible
-        _, in_features = weight.shape
+        out_features, in_features = weight.shape
         inp_shape = inp.shape
         assert inp_shape[-1] == in_features, "GEMM not possible"
 
         tp_world_size = get_distributed_world_size(tp_group)
-        ub_overlap_rs = False if tp_world_size == 1 else ub_overlap_rs
-
         backward_needs_input = is_grad_enabled and weight.requires_grad
 
         # Prepare input tensor
         # Note: Cast to expected dtype and perform tensor-parallel communication
         inputmat = inp
         inputmat_total = None
-        with_input_all_gather = parallel_mode == "column" and sequence_parallel
+        with_input_all_gather_nccl = (
+            parallel_mode == "column" and sequence_parallel and not ub_overlap_ag_fprop
+        )
         own_quantized_input = False
         if fp8:
+            if any([ub_overlap_ag_fprop, ub_overlap_rs_fprop]) and isinstance(
+                FP8GlobalStateManager.get_fp8_recipe(), BlockScaling
+            ):
+                raise NotImplementedError("Comm+GEMM overlap does not support MXFP8 block scaling")
+
```

Comment:
Same as https://github.com/NVIDIA/TransformerEngine/pull/1427#discussion_r1932928632:
```suggestion
        # Check if overlapped communication is supported
        if (
            fp8
            and (ub_overlap_ag_fprop or ub_overlap_rs_fprop)
            and not FP8GlobalStateManager.get_fp8_recipe().delayed()
        ):
            raise NotImplementedError("Comm+GEMM overlap is only supported with FP8 delayed scaling")

        # Prepare input tensor
        # Note: Cast to expected dtype and perform tensor-parallel communication
        inputmat = inp
        inputmat_total = None
        with_input_all_gather_nccl = (
            parallel_mode == "column" and sequence_parallel and not ub_overlap_ag_fprop
        )
        own_quantized_input = False
        if fp8:
```